flux/karpenter-config/node-template.yaml

```diff
@@ -0,0 +1,16 @@
+apiVersion: karpenter.k8s.aws/v1alpha1
+kind: AWSNodeTemplate
+metadata:
+  name: prowjob-node-provider
+spec:
+  securityGroupSelector:
+    kubernetes.io/cluster/TestInfraCluster: owned
+  subnetSelector:
+    kubernetes.io/cluster/TestInfraCluster: owned
```

Comment:
We should make this `TestInfraCluster` a parameter; right now it lives in CDK here https://github.com/aws-controllers-k8s/test-infra/blob/b54ff65022312f8b425279857f33891a8b542eba/infra/lib/test-ci-stack.ts#L12

How can we make it so that this doesn't need to be manually changed when we change the name of the cluster?

Reply:
Disregard this, I just noticed that you're updating it below

Reply:
I have updated the CDK to now create a new ConfigMap in the `flux-system` namespace which contains `tagKey` and `tagValue` data values. Then, when we apply the `Kustomization` which creates the `karpenter-config` resources, we use a `substituteFrom` which takes all values from that configmap and applies it into the `AWSNodeTemplate` dynamically.